lib/std/net.zig

```diff
@@ -451,7 +451,11 @@ pub fn getAddressList(allocator: *mem.Allocator, name: []const u8, port: u16) !*
             .next = null,
         };
         var res: *os.addrinfo = undefined;
-        switch (os.system.getaddrinfo(name_c.ptr, port_c.ptr, &hints, &res)) {
+        switch (os.system.getaddrinfo(
+                @ptrCast([*:0]const u8, name_c.ptr),
```

Comment:
likewise

Reply:
@andrewrk mentioned there are valid implicit casts that havent made it to the compiler, this might be one of them.

```
 error: expected type '[*:0]const u8', found '[*]u8'
        switch (os.system.getaddrinfo(name_c.ptr, port_c.ptr, &hints, &res)) {

error: expected type '[*:0]const u8', found '[]u8'
        switch (os.system.getaddrinfo(name_c, port_c, &hints, &res)) {
```

Shouldn't the parameters to getaddrinfo be null terminated?

Reply:
You still need the `.ptr`, but the `@ptrCast` shouldbn't be needed.

Reply:
Though you may need to fix e.g. the `std.cstr.addNullByte` return type too

Reply:
Of course!